examples-gallery/plot_crane_moving_a_load.py

```diff
@@ -0,0 +1,294 @@
+    velocity_constraints=speed_constr)
+
+(fr, frstar) = KM.kanes_equations(BODY, FL)
+EOM = kd.col_join(fr + frstar)
+EOM = EOM.col_join(config_constr)
+sm.pprint(EOM)
+# %%
+# Set up the optimization problem and solve it.
+# Note: While opty could calculate the reaction forces on, say, the mover,
+# (add them to the state variables), my trials showed that even with this
+# small example the calculation time increases enourmously.
+
+state_symbols = tuple((*q_ind, *q_dep, *u_ind, *u_dep))
+constant_symbols = (l, m1, m2, g)
+specified_symbols = (F,)
+h_opty = sm.symbols('h_opty')
+unknown_symbols = []
+methode = "backward euler"
+
+num_nodes = 250
+duration =(num_nodes - 1) * h_opty
+interval_value = h_opty
+
+# Specify the known system parameters.
+par_map = OrderedDict()
+par_map[l] = 5.0
+par_map[m1] = 1.0
+par_map[m2] = 10.0
+par_map[g] = 9.81
+
+t0, tf = 0.0, duration
+
+def obj(free):
+    """Minimize h_opty, the time interval between nodes."""
+    return free[-1]
+
+def obj_grad(free):
+    grad = np.zeros_like(free)
+    grad[-1] = 1.
+    return grad
+
+# start location and and final location of the load.
+anfang = 0.0
+ende = 15.0
+
+# %%
+# Set the initial and final states to form the instance constraints.
+initial_state_constraints = {xc: anfang, xl: anfang, yl: -par_map[l], q: 0.0,
+    uxc: 0.0, uxl: 0.0, uyl: 0.0, u: 0.0}
+final_state_constraints = {xc: ende, xl: ende, yl: -par_map[l], q: 0.0,
+    uxc: 0.0, uxl: 0.0, uyl: 0.0, u: 0.0}
+instance_constraints = (
+) + tuple(
+    xi.subs({t: t0}) - xi_val for xi, xi_val in
+        initial_state_constraints.items()
+) + tuple(
+    xi.subs({t: tf}) - xi_val for xi, xi_val in
+        final_state_constraints.items()
+)
+# %%
+# Forcing h_opty > 0.0 sometimes avoids negative 'solutions'. Here it also
+# seem to help with the convergence of the optimization: Different bounds,
+# e,g, h_opty = (1.e-4, 1.) gives unreasonable results.
+
+bounds = {
+    F: (-20., 20.),
+    xl: (anfang, ende),
+    xc: (anfang, ende),
+    h_opty: (1.e-5, 0.5),
+    }
+
+# Create the optimization problem.
+prob = Problem(
+    obj,
+    obj_grad,
+    EOM,
+    state_symbols,
+    num_nodes,
+    interval_value,
+    known_parameter_map=par_map,
+    instance_constraints=instance_constraints,
+    bounds=bounds,
+    integration_method=methode,
+    )
+# %%
+# Using a 'reasonable' initial guess speeds up the optimization process a lot.
+# 'Bad' initial guesses may even prevent convergence.
+i1 = [(ende - anfang) / num_nodes * i for i in range(num_nodes)]
+i2 = [0. for _ in range(num_nodes)]
+i3 = i1
+i4 = [-par_map[l] for _ in range(num_nodes)]
+i5 = [0. for _ in range(5*num_nodes)]
+i6 = [0.01]
+initial_guess = np.array(i1 + i2 + i3 + i4 + i5 + i6)
+
+# %%
+# Plot the initial guess.
+prob.plot_trajectories(initial_guess)
+
+
+# this allows to change the maximum number of iterations.
+# Standard is 3000.
+prob.add_option('max_iter', 1000)
+# Find the optimal solution.
+for _ in range(1):
```

Comment:
Remove this loop if there is only one iteration.

Ideally all examples solve with a single call to `.solve()`.